src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -55,51 +57,62 @@ Copyright (c) .NET Foundation. All rights reserved.
     <NETSdkError Condition="'@(_JitResolvedPath)'== ''"
                  ResourceName="UnableToFindResolvedPath"
                  FormatArguments="jit" />
-    
-    
-    
+
     <!-- Get the crossgen and jit path-->
     <PropertyGroup>
       <_CoreclrPath>@(_CoreclrResolvedPath)</_CoreclrPath>
-      <_JitPath>@(_JitResolvedPath)</_JitPath>
+      <JitPath>@(_JitResolvedPath)</JitPath>
       <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrPath)))</_CoreclrDir>
       <_CoreclrPkgDir>$([System.IO.Path]::Combine($(_CoreclrDir),"..\..\..\"))</_CoreclrPkgDir>
-      <_CrossgenDir>$([System.IO.Path]::Combine($(_CoreclrPkgDir),"tools"))</_CrossgenDir>
+      <CrossgenDir>$([System.IO.Path]::Combine($(_CoreclrPkgDir),"tools"))</CrossgenDir>
       <!-- TODO override with rid specific tools path for x-arch -->
-      <_Crossgen>$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen"))</_Crossgen>
-      <_Crossgen Condition="$(OS.ToUpper().Contains('WINDOWS'))">$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen.exe"))</_Crossgen>
+      <Crossgen>$([System.IO.Path]::Combine($(CrossgenDir),"crossgen"))</Crossgen>
+      <Crossgen Condition="$(OS.ToUpper().Contains('WINDOWS'))">$([System.IO.Path]::Combine($(CrossgenDir),"crossgen.exe"))</Crossgen>
     </PropertyGroup>
 
-    <NETSdkError Condition="!Exists($(_Crossgen))"
+    <NETSdkError Condition="!Exists($(Crossgen))"
                  ResourceName="UnableToFindResolvedPath"
-                 FormatArguments="$(_Crossgen)" />
+                 FormatArguments="$(Crossgen)" />
 
     <PropertyGroup>
-      <_Crossgen> $([System.IO.Path]::GetFullPath($(_Crossgen)))</_Crossgen>
+      <Crossgen> $([System.IO.Path]::GetFullPath($(Crossgen)))</Crossgen>
     </PropertyGroup>
+  </Target>
+  
+  <!--
+    ============================================================
+                                        _RunOptimizer
 
+    Start the optimization phase
+    ============================================================
+    -->
+  
+  <Target Name="_RunOptimizer"
+          DependsOnTargets="_InitializeBasicProps;
+                            _ComputeResolvedFilesToCacheTypes;
+                            _SetupStageForCrossgen"
+          Condition="$(SkipOptimization) != 'true' ">
+    
     <!-- Create the output directory where crossgened assemblies are going to be present -->
     <ItemGroup>
-      <_RuntimeOptimizedDirectories Include="@(_AssembliestoCrossGen -> '$(_RuntimeOptimizedDir)\%(RecursiveDir)')"/>
+      <_RuntimeOptimizedOutput Include="@(_ManagedResolvedFilesToOptimize -> '$(_RuntimeOptimizedDir)$(DirectorySeparatorChar)%(DestinationSubPath)')"/>
+      <_RuntimeOptimizedDirectories Include="@(_RuntimeOptimizedOutput -> '%(RootDir)%(Directory)')"/>
     </ItemGroup>
-    <MakeDir  Directories="@(_RuntimeOptimizedDirectories)"/>
+    
+    <MakeDir  Directories="@(_RuntimeOptimizedDirectories )"/>
```

Comment:
As i have stopped copying temp files, I no longer need the nugget layouts from now on, will remove these